agent/dockerclient/dockerapi/docker_client.go

```diff
@@ -337,29 +341,65 @@ func (dg *dockerGoClient) time() ttime.Time {
 // Pulls image manifest from the registry
 func (dg *dockerGoClient) PullImageManifest(
 	ctx context.Context, imageRef string, authData *apicontainer.RegistryAuthenticationData,
-) (registry.DistributionInspect, error) {
+) (registry.DistributionInspect, apierrors.NamedError) {
 	// Get auth creds
 	sdkAuthConfig, err := dg.getAuthdata(imageRef, authData)
 	if err != nil {
-		return registry.DistributionInspect{}, wrapPullErrorAsNamedError(imageRef, err)
+		return registry.DistributionInspect{}, wrapManifestPullErrorAsNamedError(imageRef, err)
 	}
 	encodedAuth, err := registry.EncodeAuthConfig(sdkAuthConfig)
 	if err != nil {
-		return registry.DistributionInspect{}, wrapPullErrorAsNamedError(imageRef, err)
+		return registry.DistributionInspect{}, wrapManifestPullErrorAsNamedError(imageRef, err)
 	}
 
-	// Get an SDK Docker Client and call Distribution API
+	// Get an SDK Docker Client
 	client, err := dg.sdkDockerClient()
 	if err != nil {
 		return registry.DistributionInspect{}, CannotGetDockerClientError{version: dg.version, err: err}
 	}
-	distInspect, err := client.DistributionInspect(ctx, imageRef, encodedAuth)
+
+	// Call DistributionInspect API with retries
+	var distInspectPtr *registry.DistributionInspect
+	err = retry.RetryNWithBackoffCtx(ctx, dg.manifestPullBackoff, maximumPullRetries, func() error {
+		distInspect, err := client.DistributionInspect(ctx, imageRef, encodedAuth)
+		if err != nil {
+			return err
+		}
+		distInspectPtr = &distInspect
+		return nil
+	})
+
 	if err != nil {
-		err = redactEcrUrls(imageRef, err)
-		return registry.DistributionInspect{}, CannotPullContainerError{err}
+		return registry.DistributionInspect{}, wrapManifestPullErrorAsNamedError(imageRef, err)
+	}
+	if ctxErr := ctx.Err(); ctxErr != nil {
+		// Context was done while waiting for a retry attempt
+		return registry.DistributionInspect{}, wrapManifestPullErrorAsNamedError(imageRef, ctxErr)
```

Comment:
I noticed for `PullImage` that if context error is due to context's deadline being exceeded, that a `DockerTimeoutError` is used instead of `CannotPullContainerError` ([ref](https://github.com/aws/amazon-ecs-agent/blob/f8c0dd0932853e338c6124ee9e38bb29d3d0fbd1/agent/dockerclient/dockerapi/docker_client.go#L429-L432)). Do we want to do that here as well for consistency (i.e., use `DockerTimeoutError` instead of `CannotPullImageManifestError` in this case)?

Reply:
Good catch! I will update it to match `PullImage`.